### ISSUE TITLE
oidc-callback infinite redirect loop - with IdentiyServer 4

### ISSUE THREAD

**Issue by DuelMonster** [perarnborg/vuex-oidc-example-nuxt#5]
Hey, hoping someone can help me out...?

I have an existing project written in Vue.js using vuex-oidc and working alongside IdentityServer 4, which I'm looking at converting to Nuxt.js for various reasons.

Initially the project works well; I get redirected to my IS4 login page, but after having logged on I get stuck in a never ending redirect loop on the oidc-callback view.  I can also see that the token is being updated on every redirect.

I assumed it was something wrong in my project but having downloaded this example, and changing the oidcSettings to point at my IS4 server, I'm getting the same result.

Anyone have any ideas, before I end up with a cold head from hair pulling?
Many Thanks

**/config/oidc.js - example**
```javascript
const IS4_DOMAIN = 'https://localhost:44351'
const NUXTJS_DOMAIN = window.location.origin

export const oidcSettings = {
  authority: IS4_DOMAIN,
  clientId: 'nuxt.js.oidc.test',
  clientSecrets: [{value: ''}],
  responseType: 'id_token token',
  scope: 'openid profile email tmr_api',
  redirectUri: NUXTJS_DOMAIN  + '/oidc-callback',
  postLogoutRedirectUri: NUXTJS_DOMAIN ,
  automaticSilentRenew: true,
  silentRedirectUri: NUXTJS_DOMAIN + '/silent-renew.html',
  filterProtocolClaims: true
}
```

**Comment by dschox** [comment]
This example uses an old version of vuex-oidc. Update the dependency and try again.
https://github.com/perarnborg/vuex-oidc-example-nuxt/issues/1#issuecomment-508372321

**Comment by DuelMonster** [comment]
> This example uses an old version of vuex-oidc. Update the dependency and try again.
> [#1 (comment)](https://github.com/perarnborg/vuex-oidc-example-nuxt/issues/1#issuecomment-508372321)

Yea, I caught that issue and updated it.  It wouldn't even run with the old dependency...

**Comment by dschox** [comment]
We had a redirect loop, too. After some digging, we found out, that the router base was the problem. We had our app running in a subfolder (eg /app/). After manually changing a line in the lib, all worked fine.
I opened an issue for that: https://github.com/perarnborg/vuex-oidc-example-nuxt/issues/6#issue-468158859
We have now moved away from this approach because of other reasons.
Maybe this helps...

**Comment by DuelMonster** [comment]
hmm, sounds promising... thnx @dschox, I'll do a little testing

**Comment by DuelMonster** [comment]
WOW, found it and it was the most silliest thing...

the `routeIsOidcCallback` function was comparing my callback path `'/oidc-callback'` with the expected path of `'/oidc-callback/'`...

Added the missing `/` to my config and all working perfectly 😁
That trailing slash must have got added at some point by IS4 I'm guessing, as I don't see vuex-oidc adding it...
Maybe the `routeIsOidcCallback` function could be a little more lenient with the trailing slash?

Thank you ever so much for the assist @dschox!  At least I shan't need to throw that comb away 😋

**Comment by dschox** [comment]
Ahh, we had that trailing slash too! :D

**Comment by perarnborg** [comment]
This issue should be fixed as of v2.0.2. Sorry for this mistake!

**Comment by DuelMonster** [comment]
Fix confirmed working.  Nice one @perarnborg 😀
